src/radical/pilot/agent/launch_method/ibrun.py

```diff
@@ -41,38 +39,52 @@ def construct_command(self, cu, launch_script_hop):
 
         slots        = cu['slots']
         cud          = cu['description']
+
         task_exec    = cud['executable']
-        task_cores   = cud['cpu_processes']  # FIXME: handle cpu_threads
         task_args    = cud.get('arguments') or []
         task_argstr  = self._create_arg_string(task_args)
+        task_env     = cud.get('environment') or dict()
+
+        n_tasks      = cud['cpu_processes']  # FIXME: handle cpu_threads
```

Comment:
We may want to move the task to `FAILED` then if threads are not handled yet.  An
```python
assert(cud.get('cpu_threads', 1) == 1), 'no thread support in IBRUN launch method'
```
would do the trick - the executor will catch the except and set the task state etc.  (Or a `ValueError` I guess...).  Or maybe I misunderstand the `FIXME` and the task will still run as expected?

Reply:
That `FIXME` was there already, I thought just to keep it for later (should I remove it?). With `cpu_threads` would be better to just ignore it (e.g., not to change the app code after switching between LMs)

Reply:
Ah, makes sense.  Can you clarify what happens in the current code when a task requests, say, 2 threads per process?  I assume ibrun will ignore that, but will also not pin the process to specific cores so that the process can indeed then create those processes?  If so, the `FIXME` can indeed be removed.

Reply:
Yeah, can confirm that threads will be ignored, and no process pinning to specific cores (offsets are at process/task level and threads are not considered)